v2/internal/crdmanagement/manager.go

```diff
@@ -154,70 +155,142 @@ func (m *Manager) FindNonMatchingCRDs(
 	return m.FindMatchingCRDs(existing, goal, invertedComparators...)
 }
 
-func (m *Manager) ApplyCRDs(
-	ctx context.Context,
+// DetermineCRDsToInstallOrUpgrade examines the set of goal CRDs and installed CRDs to determine the set which should
+// be installed or upgraded.
+func (m *Manager) DetermineCRDsToInstallOrUpgrade(
```

Comment:
Looks good, the only thing I can think of is the performance issues and delayed pod start-up due to a lot of checks and loops, which is gonna increase as we plan to add more resources. One of the improvements I can think of is, rather than storing all the CRDs by name in a Map, we can have something like `groupString:[]CRDs` map to avoid looping through each CRD and we can filter out/exclude the groups totally without iterating through each CRD, if not specified at all. The above should be good for now, and further we can do this filtering per group in parallel using semaphores.

Reply:
What's the actual runtime of this check? 
If it's on the order of milliseconds, I think we're fine. If it's a few seconds, we need to keep an eye on it - and if it's minutes we really need to fix. I suspect it's a few seconds at most.

Reply:
The actual breakdown of time from a local run on a kind cluster is:
Loading CRDs from filesystem: ~3s
Listing CRDs from k8s APIServer: ~0.5s
Filtering CRDs locally: ~0.5s
Applying CRDs: ~14s

Total time: ~18s, of which this PR adds at most ~0.5s.

Reply:
If anything, this PR is more likely to reduce startup time by skipping any CRDs that are both uninstalled and unrequested.

Reply:
That said, 14s to apply CRDs is a relatively long time. Is that single-threaded? If not, maybe we should look at introducing some concurrency.